app/services/childcare_eligibility_service.rb

```diff
@@ -0,0 +1,33 @@
+class ChildcareEligibilityService
+  class << self
+    def call(check)
+      dependants?(check) &&
+        client_eligible?(check) &&
+        (!check.partner || partner_eligible?(check))
+    end
+
+    def dependants?(check)
+      check.child_dependants
+    end
+
+    def client_eligible?(check)
+      is_student?(check.student_finance_value) || in_work?(check.employment_status, check.incomes, check.session_data)
+    end
+
+    def partner_eligible?(check)
+      is_student?(check.partner_student_finance_value) || in_work?(check.partner_employment_status, check.partner_incomes, check.session_data)
+    end
+
+    def is_student?(student_finance_amount)
+      student_finance_amount.positive?
+    end
+
+    def in_work?(employment_status, incomes, session_data)
+      if FeatureFlags.enabled?(:self_employed, session_data)
+        incomes&.any? { _1.income_type != "statutory_pay" } || false
```

Comment:
question - do we need the `|| false` - unless I'm reading it wrong does [any?](https://apidock.com/ruby/Enumerable/any%3F) evaluate as false if its an empty array.